### PR TITLE
lookup readable_graphids 1 time instead of N times in get_related_res…

### LINKED ISSUE #11457: `get_related_resources()` calls `user_can_read_graph()` multiple times for same graphs, huge slowdown
At this [section](https://github.com/archesproject/arches/blob/dev/7.6.x/arches/app/models/resource.py#L866-L867) of `get_related_resources` method (used by the resource_report API endpoint):

```
for relation in resource_relations["relations"]:
            relation = model_to_dict(relation)
            resourceid_to = relation["resourceinstanceidto"]
            resourceid_from = relation["resourceinstanceidfrom"]
            resourceinstanceto_graphid = relation["resourceinstanceto_graphid"]
            resourceinstancefrom_graphid = relation["resourceinstancefrom_graphid"]

            if (
                resourceid_to not in restricted_instances
                and resourceid_from not in restricted_instances
                and user_can_read_graph(user, resourceinstanceto_graphid)
                and user_can_read_graph(user, resourceinstancefrom_graphid)
            ):
            ...

```

the `user_can_read_graph` permissions method makes an ORM query every time it's called, even on relations shows graph we've already check. This can slow things down significantly, to the point where a single report takes up to 2 minutes to load. The slowdown is especially noticeable for resources with 50+ relations

### PR DESCRIPTION
**Pull request by whatisgalen**
…ources, re #11457

### Types of changes

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change

This PR replaces repeated use of `user_can_read_graph()` for each related resource in a report with a single lookup for a `readable_graphids` set to check against. Functionally nothing else has changed.

### Issues Solved

Closes #11457 

### Checklist

-   I targeted one of these branches:
    - [x] dev/7.6.x (under development): features, bugfixes not covered below
    - [ ] dev/7.5.x (main support): regressions, crashing bugs, security issues, major bugs in new features
    - [ ] dev/6.2.x (extended support): major security issues, data loss issues
-   [ ] I added a changelog in arches/releases
-   [ ] I submitted a PR to arches-docs (if appropriate)
-   [ ] Unit tests pass locally with my changes
-   [ ] I added tests that prove my fix is effective or that my feature works
-   [ ] My test fails on the target branch

#### Accessibility Checklist

[Developer Guide](https://arches.readthedocs.io/en/stable/developing/advanced/accessibility/)

| Topic            | Changed | Retested |
| ---------------- | ------- | -------- |
| Color contrast   |         |          |
| Form fields      |         |          |
| Headings         |         |          |
| Links            |         |          |
| Keyboard         |         |          |
| Responsive Design|         |          |
| HTML validation  |         |          |
| Screen reader    |         |          |

#### Ticket Background
*   Sponsored by: 
*   Found by: @ 
*   Tested by: @ 
*   Designed by: @

### CODE REVIEW

File: arches/app/search/components/term_filter.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def append_dsl(self, search_query_object, **kwargs):
         permitted_nodegroups = kwargs.get("permitted_nodegroups")
         include_provisional = kwargs.get("include_provisional")
         search_query = Bool()
-        querystring_params = kwargs.get("querystring", "")
+        querystring_params = kwargs.get("querystring", "[]")
```
Comment (apeters) [inline arches/app/search/components/term_filter.py:35]: this doesn't really belong in this PR

Comment (whatisgalen) [reply arches/app/search/components/term_filter.py]: removed in latest

**Review by apeters — APPROVED** [review]
looks good

**Comment by whatisgalen** [comment]
I'm feeling slightly cheated given that this PR was improving test coverage up until #11460 was merged in via [4a18954](https://github.com/archesproject/arches/pull/11458/commits/4a18954e059ca66744402f180074f80bf067bc78). Wondering if I can get a pass?

**Comment by chiatt** [comment]
> I'm feeling slightly cheated given that this PR was improving test coverage up until #11460 was merged in via [4a18954](https://github.com/archesproject/arches/pull/11458/commits/4a18954e059ca66744402f180074f80bf067bc78). Wondering if I can get a pass?

Yeah - I think that's fine
